src/clusterd/src/bin/clusterd.rs

```diff
@@ -76,6 +76,9 @@ struct Args {
         default_value = "127.0.0.1:6878"
     )]
     internal_http_listen_addr: SocketAddr,
+    /// The hostname of this process, for GRPC request validation.
+    #[clap(long, env = "GRPC_HOSTNAME", value_name = "NAME")]
+    grpc_hostname: Option<String>,
```

Comment:
We would have had the option to always use the `HOSTNAME` env variable directly and then set `HOSTNAME` instead of `CLUSTERD_GRPC_HOSTNAME` in `mzcompose/services/clusterd.py`. Then we wouldn't have to do anything special when running inside of k8s, but we'd have to somehow deal with opting out of the check when running locally (and maybe in other contexts too). Explicitly opting into the check seems more predictable.

Reply:
Agreed—I like this.